### PR TITLE
feat: set env for server

### PR DESCRIPTION
**Pull request by alvinhui**
@chenbin92 

Should we add a method call `execa` for project, then adapter using `project.execa`  to execute command?

Following benefits:

- Developer don't need to declare env explicitly;
- We can do some special processing in this way, such as printing global logs.

### CODE REVIEW

File: packages/iceworks-server/src/lib/plugin/project-manager/app.ts [hunk 1]
```diff
@@ -69,7 +71,26 @@ class Project implements IProject {
...
+    const PATH = pathKey();
+
+    const pathEnv = [
+      env[PATH],
+    ];
+
+    // for electron 
+    const resourcesPath = process['resourcesPath']; // eslint-disable-line
+    if (resourcesPath) {
+      path.join(resourcesPath, 'bin');
```
Comment (chenbin92) [inline packages/iceworks-server/src/lib/plugin/project-manager/app.ts]: 这里不用赋值吗

File: packages/iceworks-server/src/lib/plugin/project-manager/app.ts [hunk 2]
```diff
@@ -69,7 +71,26 @@ class Project implements IProject {
...
+    }
+
+    if (os.type() === 'Darwin') {
+      pathEnv.push('/usr/local/bin');
+    }
+
+    env[PATH] = pathEnv.join(path.delimiter);
+
+    return env;
```
Comment (chenbin92) [inline packages/iceworks-server/src/lib/plugin/project-manager/app.ts]: 之前讨论的是 iceworks cli 运行在用户的机器上，应该就不存在环境问题。现在引入 env 这个概念是为兼容 Electron ？

File: packages/iceworks-server/src/lib/plugin/project-manager/app.ts (hunk 1 above)
Comment (alvinhui) [reply packages/iceworks-server/src/lib/plugin/project-manager/app.ts]: fixed

File: packages/iceworks-server/src/lib/plugin/project-manager/app.ts (hunk 2 above)
Comment (alvinhui) [reply packages/iceworks-server/src/lib/plugin/project-manager/app.ts]: 是为了兼容 Electron 的场景

**Comment by chenbin92** [comment]
> @chenbin92
> 
> Should we add a method call `execa` for project, then adapter using `project.execa` to execute command?
> 
> Following benefits:
> 
> * Developer don't need to declare env explicitly;
> * We can do some special processing in this way, such as printing global logs.

这里需要讨论下，什么样的方法适合挂载在 project，需要定义清楚，目前来看只是基于现有场景考虑了 execa 的 env，如果不挂载通过暴露 env 也是一种做法，可以另外提一个 issue ，不在这个 MR  实现

**Pull request merged**